### DISCUSSION TITLE
prophecy.js 可以考虑在一天之内针对同一个QQ号的查询只返回同样的结果

### DISCUSSION THREAD

**Discussion by mark9804** [Arondight/Adachi-BOT#469] [Ideas]
https://github.com/Arondight/Adachi-BOT/blob/5c703fd4a923125e16712e45996d227b649194f8/src/plugins/tools/prophecy.js#L3-L7
例如，`getRandomInt`这里，如果能传递一个作为可选参数的`seed`，
那么使用`QQ号+今天的日期`（例：`1111111120211205`）作为种子，应该就能实现“在同一天针对同一个QQ账号返回同一个求签结果”的功能
https://github.com/Arondight/Adachi-BOT/blob/5c703fd4a923125e16712e45996d227b649194f8/src/plugins/tools/prophecy.js#L4

**Comment by Arondight** [comment]
这个是我个人的爱好，似乎又不值得为相同与否做个开关，就暂且这样了

**Comment by mark9804** [comment]
虽然我原本的想法是不给开关直接做成一天之内结果相同，那这样的话我仿着之前的代码加上开关吧，到时候提一个PR过来

**Reply by Arondight** [reply]
不要加开关吧，我的意思是这种小功能都加开关，那有点滥用配置的嫌疑了，配置文件也不是越长越好

**Reply by mark9804** [reply]
刚刚看了一下，就算想做这个小功能，JS的`Math.random()`也没有原生的设置种子的方式，为了这个引入`seedrandom` npm 包或者自己手撸一个好像对于这个功能来说又太奢侈了

**Reply by mark9804** [reply]
另一个选择是 [npm xorshift](https://www.npmjs.com/package/xorshift) ，自己写的话不管是线性同余法还是梅森旋转法都有一个以上的不知道来历的魔法数字，我感觉可能老大哥不太愿意看到不知道来历的数字出现在项目里

**Reply by Arondight** [reply]
不需要种子啊，你都组合出一个按日变动的串了，还要随机数干啥，直接把串丢给 api ，建议用日期加（数学意义上）id

**Reply by Arondight** [reply]
> 不太愿意看到不知道来历的数字出现在项目里

我是不太想在配置文件中加这么一个开关而已，代码上我倒是没啥洁癖，至少没啥很坚持的洁癖，因为我觉得这个功能似乎……太无足轻重了
